Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -863,15 +870,49 @@ void ExecuteCommand(ReplyType reply_type)
     case 0x00:  // Read Sector
     {
       const u64 dvd_offset = static_cast<u64>(s_DICMDBUF[1]) << 2;
-
       INFO_LOG_FMT(
           DVDINTERFACE,
           "Read: DVDOffset={:08x}, DMABuffer = {:08x}, SrcLength = {:08x}, DMALength = {:08x}",
           dvd_offset, s_DIMAR, s_DICMDBUF[2], s_DILENGTH);
 
       if (s_drive_state == DriveState::ReadyNoReadsMade)
         SetDriveState(DriveState::Ready);
-
+      if (DVDThread::GetDiscType() == DiscIO::Platform::Triforce)
+      {
+        if ((dvd_offset & 0x80000000) != 0)
+        {
+          switch (dvd_offset)
+          {
+          // Media board status (1)
+          case 0x80000000:
+            Memory::Memset(s_DIMAR, 0, s_DICMDBUF[2]);
```

Comment:
should `s_DICMDBUF[2]` have any bounds/alignment applied throughout here?

Reply:
What do you mean?

Reply:
`s_DICMDBUF[2]` is a `u32` that can be fully controlled by software sending the cmd, so probably it is at least limited to some upper bound (either based on target buffer size or just some bitwise truncation). otherwise it could cause a 4GB write through `s_DIMAR` (and well, maybe that _does_ happen, idk how this device behaves).